content/doc/book/pipeline/pipeline-best-practices.adoc

```diff
@@ -0,0 +1,57 @@
+---
+layout: section
+title: Pipeline Best Practices
+---
+ifdef::backend-html5[]
+:notitle:
+:description:
+:author: Alex Taylor
+:email: jenkinsci-docs@googlegroups.com
+:sectanchors:
+ifdef::env-github[:imagesdir: ../resources]
+ifndef::env-github[:imagesdir: ../../resources]
+:hide-uri-scheme:
+:toc:
+endif::[]
+
+= General
+== Pipeline functions as Glue
+Make sure to use pipeline steps as a way of wiring together a set of actions rather than the main functionality. This means that rather than relying on pipeline functionality(groovy or pipeline steps) to drive the build process forward, it is better to use single steps(such as `sh`) to accomplish multiple parts of the build. The less pipeline functions used, the less memory will consumed up on master. Think of pipeline as a tool to accomplish a build rather than it being the core of that build.
```

Comment:
```suggestion
Make sure to use pipeline steps as a way of wiring together a set of actions rather than the main functionality.
This means that rather than relying on pipeline functionality(groovy or pipeline steps) to drive the build process forward, it is better to use single steps(such as `sh`) to accomplish multiple parts of the build.
The less pipeline functions used, the less memory will consumed up on master.
Think of pipeline as a tool to accomplish a build rather than it being the core of that build.
```